sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -289,6 +289,10 @@ class InvalidUnsafeRowException(error: String)
     "among restart. For the first case, you can try to restart the application without " +
     s"checkpoint or use the legacy Spark version to process the streaming state.\n$error", null)
 
+class InvalidStateStoreProvider
+  extends RuntimeException("The streaming query failed because the given StateStoreProvider " +
+    "does not extend org.apache.spark.sql.execution.streaming.state.StateStoreProvider")
```

Comment:
Can we make sure that we are using the new error class framework here ?

cc - @HeartSaVioR